src/main/java/com/minecraftabnormals/abnormals_core/core/api/conditions/ConfigValueCondition.java

```diff
@@ -0,0 +1,128 @@
+
+/**
+ * A condition that checks against the values of config values annotated with {@link ConfigKey}.
+ *
+ * <p>To make your mod's config values compatible with it, annotate them with {@link ConfigKey} taking in the string
+ * value that should be used to deserialize the field, then call {@link DataUtil#registerConfigCondition(String, Object...)}
+ * in the mod setup, passing your mod id as the first parameter, and the config objects with the
+ * {@link ForgeConfigSpec.ConfigValue} instances that should be checked against as the second.</p>
+ *
+ * <p>For a condition with type {@code "[modid]:config"}, it takes the arguments:
+ * <ul>
+ *   <li>{@code value}      - the name of the config value to check against, defined by its corresponding {@link ConfigKey} annotation</li>
+ *   <li>{@code predicates} - an array of JSON objects that deserialize to an {@link IConfigPredicate}, which prevent
+ *                            the condition from passing if one of more of yhem return false. Optional if {@code value}
+ *                            maps to a boolean {@link ForgeConfigSpec.ConfigValue}.</li>
+ *   <li>{@code inverted}   - whether the condition should be inverted, so it will pass if {@code predicates} return false instead.
+ *                            Inclusion is optional.</li>
+ * </ul></p>
+ *
+ * @see DataUtil#registerConfigCondition(String, Object...)
+ * @author abigailfails
+ *
+ */
+public class ConfigValueCondition implements ICondition {
+    private final ForgeConfigSpec.ConfigValue<?> value;
+    private final String valueID;
+    private final Map<IConfigPredicate, Boolean> predicates;
+    private final boolean inverted;
+    private final ResourceLocation location;
+
+    public ConfigValueCondition(ResourceLocation location, ForgeConfigSpec.ConfigValue<?> value, String valueID, Map<IConfigPredicate, Boolean> predicates, boolean inverted) {
+        this.location = location;
+        this.value = value;
+        this.valueID = valueID;
+        this.predicates = predicates;
+        this.inverted = inverted;
+    }
+
+
+    @Override
+    public ResourceLocation getID() {
+        return this.location;
+    }
+
+    @Override
+    public boolean test() {
+        boolean returnValue;
+        if (predicates.size() > 0) {
+            returnValue = this.predicates.keySet().stream().allMatch(c -> this.predicates.get(c) != c.test(value));
+        } else if (value.get() instanceof Boolean) {
+            returnValue = (Boolean) value.get();
+        } else throw new IllegalStateException("Predicates required for non-boolean ConfigValue, but none found");
+        return this.inverted != returnValue;
+    }
+
+    public static class Serializer implements IConditionSerializer<ConfigValueCondition> {
+        public static final Hashtable<ResourceLocation, IConfigPredicateSerializer<?>> CONFIG_PREDICATE_SERIALIZERS = new Hashtable<>();
+        private final Hashtable<String, ForgeConfigSpec.ConfigValue<?>> CONFIG_VALUES;
```

Comment:
Why is this statically styled?
This field is also threadsafe as just a `HashMap` because it's set per instance and not modified on multiple threads.

Reply:
It used to be static until I realised duplicates should be perfectly allowable for different mod config types. I thought that was constant styling, and I guess it should be immutable as there's only one instance per class now.